File: Arcade/TheCore/Java/ArrayPacking.java
final class ArrayPacking {
	
	int arrayPacking(int[] a) {
	    String bits = "";
	    
	    for(int i = a.length - 1; i >= 0; i--) {
	        String _bin = Integer.toString(a[i], 2);
	        bits += "0".repeat(8 - _bin.length()) + _bin;
	    }
	        
	    return Integer.parseInt(bits, 2);
	}
}

File: Arcade/TheCore/Kotlin/arrayPacking.kt
fun arrayPacking(a: MutableList<Int>): Int {
    return Integer.parseInt(
        a.map { it.toString(2) }
        .reversed()
        .joinToString(
            separator = "",
            transform = { "0".repeat(8 - it.length) + it }
        ), 2
    )
}